Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/FacepunchMultiplayerMenu/FacepunchMultiplayerMenu.cs

```diff
@@ -0,0 +1,210 @@
+﻿#if FACEPUNCH_STEAMWORKS
+using BeardedManStudios.Forge.Networking;
+using BeardedManStudios.Forge.Networking.Unity;
+using BeardedManStudios.Forge.Logging;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+using Steamworks;
+using System.Threading.Tasks;
+
+
+public class FacepunchMultiplayerMenu : MonoBehaviour
+{
+	public const int MAXIMUM_SERVER_SLOTS = 64;
+
+	public bool dontChangeSceneOnConnect = false;
+	public GameObject networkManager = null;
+	public GameObject[] toggledButtons;
+	public Steamworks.Data.Lobby lobbyToJoin = default;
+	public bool useMainThreadManagerForRPCs = true;
+	public bool useInlineChat = false;
+
+	private NetworkManager mgr = null;
+	private NetWorker server;
+	private Steamworks.Data.Lobby lobby;
+	private List<Button> _uiButtons = new List<Button>();
+
+	private void Start()
+	{
+		for (int i = 0; i < toggledButtons.Length; ++i)
+		{
+			Button btn = toggledButtons[i].GetComponent<Button>();
+			if (btn != null)
+				_uiButtons.Add(btn);
+		}
+
+		if (useMainThreadManagerForRPCs)
+			Rpc.MainThreadRunner = MainThreadManager.Instance;
+	}
+
+	/// <summary>
+	/// Connects to the currently selected Steam Lobby
+	/// Lobby selection is made within FacepuncJoinMenu.cs
+	/// </summary>
+	public void Connect()
+	{
+		if (lobbyToJoin.Id > 0)
+		{
+			ConnectToLobbyAsync();
+		}
+
+		// TODO:  Raise a dialog to inform of the failed connection attempt
```

Comment:
This todo should probably be handled in an example project and in this implementation part should maybe raise an exception or trigger an event?

Reply:
I think it should trigger an event, and hence, could be utilized in every project

Reply:
Would a LogWarning do for this? This is kind of a stripped down example menu like the others in forge, so should be only there as a guide for new users anyway.

Reply:
Maybe add in some handling that calls a LogWarning for now with a small comment that this is the place where someone would add functionality to show the message to the user as a popup for example?